### PR TITLE
HttpServer performance improvement

### LINKED ISSUE #6: TreeMap
Implement a class for efficient lookup based on path prefixes. Its primary use will be in HttpServer for faster matching of urls to handlers.

### PR DESCRIPTION
**Pull request by tethal**
@davidnich, could you do a review please?

### CODE REVIEW

File: qlib/HttpServer.qm [hunk 1]
```diff
@@ -119,6 +119,15 @@ log("started listener on %s", lh.desc);
 
     @section http_relnotes HttpServer Module Release Notes
 
+    @subsection http0312 HttpServer 0.3.12
```
Comment (davidnich) [inline qlib/HttpServer.qm]: 0.3.11 has not yet been released - please move the changes down to that version

**Comment by davidnich** [comment]
please add a note in doxygen/lang/900_release_notes.dox.tmpl for the new feature as well

**Comment by davidnich** [comment]
now do we delete the branch or what?

**Comment by tethal** [comment]
Yes. There may be some cases where we merge an incomplete feature into master, but I guess it'd be very rare.

**Pull request merged**